### PR TITLE
Api design

### PR DESCRIPTION
**Pull request by vasanthaganeshk**
About the sky, The bot is going to send its move as JSON object(not necessarily just wishful thinking) to the engine via a message queue(zmq?). So I thought sky can be definition of what kind object the engine accepts. I do not understand what kind of function calls sky can have. Please explain.

### CODE REVIEW

**Comment by oyeb** [comment]
First, I'm sorry to have confused all of you, I was confused myself on what I wanted -- I was not looking for an API. I wanted us to write down all the functions (which we'll implement ASAP) in the engine that involve responses/actions in bot or the game.
This way we'd know what kind of responses and actions a bot, or a game would _have to_ implement, and we would argue whether this design is scalable and decoupled enough from choice of game.
Is this ok? Checkout `sky.md`.

Second, the messages between `game` and `bot` are not inspected by the engine. And hence they can be of any format.
The messages between `game` and `engine` or `bot` and `engine` need to conform to JSON (or whatever _we want_), as these _will be inspected_ by the engine.

**Comment by vasanthaganeshk** [comment]
For the logs, can we do it like this: Don't give the access of the log to the bot (we can restrict that particular system call 'open()'? so that the bot doesn't go on opening fds). Instead we can make the engine do this task, i.e. get the "move" object from message queue and write it to the log. This log will be maintained for each bot. We can also have another log file for logging each new "game-state" object for replaying the game on the website.

File: sky.md [hunk 1]
```diff
@@ -42,6 +42,9 @@ On the other hand, a bot might _actually_ be unresponsive, and the scores awarde
 :   When a bot is unable to make a move it is said to be "defeated". Engine inspects the `game-state` at each turn to kill bots that have been "defeated".
 Engine appends a game-summary to the `logs` (when `append_logs`) is called.
 
+**init_syscall_filter()**
```
Comment (oyeb) [inline sky.md:45]: When would this be called? Would `init_bot` call this?

**Comment by oyeb** [comment]
@vasanthaganeshk Yes. That's the plan - log everything. All `game-state`s when replayed together are nothing but the game replay. And we certainly block `open`, `fork`, `exec*` (and what else??).

Let's do what I did last year. The engine wrote all the log files and the bots did not have access to these _files_.
Along with `game-state` updates at each turn, they also got a `logs` object. This object contained ***only some of the log items*** created by the game only for that turn. For ex, we should include the 

* `error`, a textual description of it, etc.
* `info`,
  - any info that would help our participant read the game logs better.
  - any info that the bot could optionally use. Last year, when a bot lost/captured a node in the map, we added an info event. Ofcourse the bot can infer that from the _change_ in it's game state, we still provided this. Note that game-wide events should be part of the `game-state`, only bot specific ones appear here.

File: sky.md (hunk 1 above)
Comment (vasanthaganeshk) [reply sky.md:45]: yes we can call it inside **init_bot()** but it would be good if we define in a separate module.

**Comment by vasanthaganeshk** [comment]
I added the method for continuous and discrete games for creating the next state from the previous state.

**Pull request merged**